src/Bicep.Core/Emit/ScopeHelper.cs

```diff
@@ -130,7 +151,20 @@ public class ScopeData
                         return null;
                     }
 
-                    return new ScopeData { RequestedScope = ResourceScope.Resource, ResourceScopeSymbol = targetResourceSymbol };
+                    return new ScopeData { RequestedScope = ResourceScope.Resource, ResourceScopeSymbol = targetResourceSymbol, IndexExpression = indexExpression };
+
+                case { } when scopeSymbol is ModuleSymbol targetModuleSymbol:
```

Comment:
Clearly my comment "type validation should have already caught this" was not accurate 😄 

Maybe we should just always log an error if we don't match what we expect, even if it results in multiple error messages?

Reply:
thinking something like:
```
if (scopeType is IScopeReference scopeReference)
{
    logInvalidScopeFunc(scopeProperty.Value, scopeReference.Scope, supportedScopes);
    return null;
}
```

Reply:
Yeah I'm torn on this. In many other places, we tend to avoid adding too many errors inline with Anders' suggestion to avoid errors caused by other errors. On the other hand, this would prevent validation bugs in the future or at least reduce the chance.